Src/ILGPU/IR/Serialization/BinaryIRWriter.cs

```diff
@@ -0,0 +1,62 @@
+﻿// ---------------------------------------------------------------------------------------
+//                                        ILGPU
+//                        Copyright (c) 2024 ILGPU Project
+//                                    www.ilgpu.net
+//
+// File: BinaryIRWriter.cs
+//
+// This file is part of ILGPU and is distributed under the University of Illinois Open
+// Source License. See LICENSE.txt for details.
+// ---------------------------------------------------------------------------------------
+
+using System;
+using System.IO;
+using System.Text;
+
+namespace ILGPU.IR.Serialization
+{
+    /// <summary>
+    /// Wrapper class around <see cref="BinaryWriter"/> for serializing IR types and values. 
+    /// </summary>
+    public sealed partial class BinaryIRWriter : IIRWriter
```

Comment:
Nit: please use our multithreading friendly `DisposeBase` class as a base class instead of implementing `IDisposable` manually.